app/scripts/lib/multichain-api/scope/assert.ts

```
@@ -0,0 +1,103 @@
+  }: {
+    findNetworkClientIdByChainId: (chainId: Hex) => NetworkClientId;
+    getInternalAccounts: () => InternalAccount[];
+  },
+) => {
+  const { methods, notifications, accounts } = scopeObject;
+  if (!isSupportedScopeString(scopeString, findNetworkClientIdByChainId)) {
+    throw new EthereumRpcError(5100, 'Requested chains are not supported');
+  }
+
+  // Needs to be split by namespace?
+  const allMethodsSupported = methods.every((method) =>
+    validRpcMethods.includes(method),
+  );
+  if (!allMethodsSupported) {
+    // not sure which one of these to use
+    // When provider evaluates requested methods to not be supported
+    //   code = 5101
+    //   message = "Requested methods are not supported"
+    // When provider does not recognize one or more requested method(s)
+    //   code = 5201
+    //   message = "Unknown method(s) requested"
+
+    throw new EthereumRpcError(5101, 'Requested methods are not supported');
+  }
+
+  if (
+    notifications &&
+    !notifications.every((notification) =>
+      isSupportedNotification(notification),
+    )
+  ) {
+    // not sure which one of these to use
+    // When provider evaluates requested notifications to not be supported
+    //   code = 5102
+    //   message = "Requested notifications are not supported"
+    // When provider does not recognize one or more requested notification(s)
+    //   code = 5202
+    //   message = "Unknown notification(s) requested"
+    throw new EthereumRpcError(
+      5102,
+      'Requested notifications are not supported',
+    );
+  }
+
+  if (accounts) {
+    const accountsSupported = accounts.every((account) =>
+      isSupportedAccount(account, getInternalAccounts),
+    );
+
+    if (!accountsSupported) {
+      // TODO: There is no error code or message specified in the CAIP-25 spec for when accounts are not supported
+      // The below is made up
+      throw new EthereumRpcError(5103, 'Requested accounts are not supported');
+    }
```

As I said [here](https://github.com/MetaMask/metamask-extension/pull/25708/files#r1670939838) I don't think should throw an error for "unsupported" accounts...